File: sdks/python/scripts/install-smoke.py
#!/usr/bin/env python3
"""Install a published wheel/sdist and execute one real call."""

from __future__ import annotations

import argparse
import subprocess
import sys


def main() -> int:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("--index-url", required=True)
    parser.add_argument("--package", default="solvapay")
    parser.add_argument("--version", required=True)
    parser.add_argument("--extra-index-url", default="")
    args = parser.parse_args()

    cmd = [
        sys.executable,
        "-m",
        "pip",
        "install",
        "--no-cache-dir",
        "--index-url",
        args.index_url,
        f"{args.package}=={args.version}",
    ]
    if args.extra_index_url:
        cmd.extend(["--extra-index-url", args.extra_index_url])
    subprocess.check_call(cmd)
    if args.package == "solvapay":
        script = (
            "import solvapay\n"
            "print('install-smoke:', solvapay.version())\n"
            "print('install-smoke-build:', solvapay.native_build_info())\n"
        )
    else:
        script = (
            "import solvapay_mcp\n"
            "print('install-smoke:', getattr(solvapay_mcp, '__version__', 'ok'))\n"
        )
    subprocess.check_call([sys.executable, "-c", script])
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: sdks/ruby/scripts/install-smoke.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Install a published gem from a host and execute one real call.

require "optparse"

host = nil
name = "solvapay"
version = nil

OptionParser.new do |opts|
  opts.on("--host HOST", "RubyGems host") { |v| host = v }
  opts.on("--name NAME", "Gem name") { |v| name = v }
  opts.on("--version VERSION", "Gem version") { |v| version = v }
end.parse!

abort "install-smoke: --host and --version are required" if host.nil? || version.nil?

cmd = ["gem", "install", name, "--version", version, "--clear-sources", "--source", host]
warn cmd.join(" ")
system(*cmd) || abort("install-smoke: gem install failed")

code = <<~RUBY
  require "#{name == 'solvapay' ? 'solvapay' : 'solvapay/mcp'}"
  puts "install-smoke: #{name == 'solvapay' ? 'SolvaPay.version' : 'SolvaPay::Mcp::VERSION'}"
  puts #{name == 'solvapay' ? 'SolvaPay.version' : 'SolvaPay::Mcp::VERSION'}
RUBY
system("ruby", "-e", code) || abort("install-smoke: runtime call failed")
